src/WooCommerce.php

```diff
@@ -64,6 +65,15 @@ public static function getProductDetails($product_id = 0, $primary_category = TR
       $category = static::getProductCategoriesParentsList($product_id);
     }
 
+    if (empty($category)) {
+      if ($primary_category) {
+        $category = static::getProductCategoryParents(static::getProductPrimaryCategoryId($parent_id ?: $product_id), '/');
+      }
+      else {
+        $category = static::getProductCategoriesParentsList($parent_id);
+      }
+    }
+
```

Comment:
Not trying to be a pain in here, but to me it seems like there might be too much duplication, so a suggestion from my side is combine this block and the previous into:
```suggestion
    if ($primary_category) {
      if (!$category = static::getProductCategoryParents(static::getProductPrimaryCategoryId($product_id), '/') && $parent_id) {
        $category = static::getProductCategoryParents(static::getProductPrimaryCategoryId($parent_id), '/');
      }
    }
    else {
      if (!$category = static::getProductCategoriesParentsList($product_id) && $parent_id) {
        $category = static::getProductCategoriesParentsList($parent_id);
      }
    }

```
Feel free to ignore if you think that is too difficult to read.